utils/checks.stanza

```diff
@@ -1728,23 +1768,25 @@ public defn propagate-power-states (module:JITXObject) :
   ; DFS walk of the netlist, to update power-states of 
   ; connected power-pins 
   val visited = HashSet<JITXObject>()
-  defn visit (pin:JITXObject, power-states:Tuple<PowerState>) :
+  defn visit-pins (pin:JITXObject, power-states-tuple:Tuple<PowerState>) :
     if not visited[pin] :
       add(visited, pin)
       if not has-property?(pin.power-states) :
-        set-property(pin, `power-states, power-states)
+        set-property(pin, `power-states, power-states-tuple)
         
       if has-property?(pin.power-pin) or
          has-property?(pin.power-supply-pin) :
-        do(visit{_, power-states}, connected-pins(pin))
+        do(visit-pins{_, power-states-tuple}, connected-pins(pin))
 
-      val instance = containing-instance!(pin)
-      if has-property?(instance.power-supply-component) and
-         property(instance.power-supply-component) : 
-        do(visit{_, power-states}, pins(instance))
+  defn visit-power-supply-components (pin:JITXObject, power-states-tuple:Tuple<PowerState>) :
+    val instance = containing-instance!(pin)
+    if has-property?(instance.power-supply-component) and
+        property(instance.power-supply-component) : 
+      do(visit-pins{_, power-states-tuple}, pins(instance))
 
   ; propagate the power-states to connected power-pins
-  do(visit{_0, property(_0.power-states)}, roots)
+  do(visit-pins{_0, property(_0.power-states)}, roots)
+  do(visit-power-supply-components{_0, property(_0.power-states)}, roots)
```

Comment:
This change looks out of scope of the rest of the PR which only touches passives and OP checks, it may be contributing to the failure seen

Reply:
Yes, I overloaded the PR in the rush to get it done. This was an attempt to fix another propagation issue.